### PR TITLE
Add spill stats reporting to PeriodicStatsReporter

### PR DESCRIPTION
**Pull request by tanjialiang**
Add additional stats reporting for spill in PeriodicStatsReporter

### CODE REVIEW

File: velox/common/base/PeriodicStatsReporter.cpp [hunk 1]
```diff
@@ -228,4 +232,13 @@ void PeriodicStatsReporter::reportCacheStats() {
   lastCacheStats_ = cacheStats;
 }
 
+void PeriodicStatsReporter::reportSpillStats() {
+  const auto spillMemoryStats = velox::memory::spillMemoryPool()->stats();
+  LOG(INFO) << "Spill memory usage: current["
```
Comment (xiaoxmeng) [inline velox/common/base/PeriodicStatsReporter.cpp:237]: nit: consider to have VELOX_STATS_LOG with [STATS] prefix to ease log search. Thanks!

File: velox/common/base/PeriodicStatsReporter.cpp [hunk 2]
```diff
@@ -228,4 +232,13 @@ void PeriodicStatsReporter::reportCacheStats() {
   lastCacheStats_ = cacheStats;
 }
 
+void PeriodicStatsReporter::reportSpillStats() {
+  const auto spillMemoryStats = velox::memory::spillMemoryPool()->stats();
```
Comment (xiaoxmeng) [inline velox/common/base/PeriodicStatsReporter.cpp:236]: Shall we just log delta spill stats? stats->toString() which shall include the peak and current memory usage? Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@tanjialiang LGTM % minors. Thanks!

File: velox/common/base/PeriodicStatsReporter.cpp (hunk 2 above)
Comment (tanjialiang) [reply velox/common/base/PeriodicStatsReporter.cpp:236]: This is memory pool stats which is snapshot. Sure we can have stats->toString()

File: velox/common/base/PeriodicStatsReporter.cpp (hunk 1 above)
Comment (tanjialiang) [reply velox/common/base/PeriodicStatsReporter.cpp:237]: I did not find VELOX_STATS_LOG

File: velox/common/base/PeriodicStatsReporter.cpp [hunk 3]
```diff
@@ -228,4 +233,11 @@ void PeriodicStatsReporter::reportCacheStats() {
   lastCacheStats_ = cacheStats;
 }
 
+void PeriodicStatsReporter::reportSpillStats() {
+  const auto spillMemoryStats = spillMemoryPool_->stats();
+  LOG(INFO) << "Spill memory usage: " << spillMemoryPool_->toString();
```
Comment (xiaoxmeng) [inline velox/common/base/PeriodicStatsReporter.cpp:238]: Oh, it is better to just report the two memory stats. I was thinking to log the spill stats from globalSpillStats()

**Pull request closed without merging**